File: app/back/src/main/java/net/seijishikin/jp/normalize/manage/kanrensha/dto/riyousha/SearchRiyoushaAllCapsuleDto.java
package net.seijishikin.jp.normalize.manage.kanrensha.dto.riyousha;

import java.io.Serializable;

import net.seijishikin.jp.normalize.common_tool.dto.NaturalTextSearchPagingCapsuleDto;

/**
 * 利用者検索条件Dto
 */
public class SearchRiyoushaAllCapsuleDto extends NaturalTextSearchPagingCapsuleDto // NOPMD DataClass
        implements Serializable {

    /** Serialize id */
    private static final long serialVersionUID = 1L;

    /** APIユーザ検索フラグ */
    private Boolean isPartnerApiSearch = INIT_BOOLEAN;

    /** 運営者検索フラグ */
    private Boolean isManagerSearch = INIT_BOOLEAN;

    /** 管理者検索フラグ */
    private Boolean isAdminSearch = INIT_BOOLEAN;

    /**
     * APIユーザ検索実行フラグを取得する
     *
     * @return APIユーザ検索
     */
    public Boolean getIsPartnerApiSearch() {
        return isPartnerApiSearch;
    }

    /**
     * APIユーザ実行フラグ検索を設定する
     *
     * @param isPartnerApiSearch APIユーザ検索
     */
    public void setIsPartnerApiSearch(final Boolean isPartnerApiSearch) {
        this.isPartnerApiSearch = isPartnerApiSearch;
    }

    /**
     * 運営者検索実行フラグを取得する
     *
     * @return 運営者検索実行フラグ
     */
    public Boolean getIsManagerSearch() {
        return isManagerSearch;
    }

    /**
     * 運営者検索実行フラグを設定する
     *
     * @param isManagerSearch 運営者検索実行フラグ
     */
    public void setIsManagerSearch(final Boolean isManagerSearch) {
        this.isManagerSearch = isManagerSearch;
    }

    /**
     * SE権限検索実行フラグを取得する
     *
     * @return SE権限検索実行フラグ
     */
    public Boolean getIsAdminSearch() {
        return isAdminSearch;
    }

    /**
     * SE権限検索実行フラグを設定する
     *
     * @param isAdminSearch SE権限検索実行フラグ
     */
    public void setIsAdminSearch(final Boolean isAdminSearch) {
        this.isAdminSearch = isAdminSearch;
    }
}

File: app/front/src/components/main/dto/riyousha/searchRiyoushaAllCapsuleDto.ts
import { FrameworkPagingDto, type FrameworkPagingDtoInterface } from "seijishikin-jp-normalize_common-tool";

interface SearchRiyoushaAllCapsuleDtoInterface extends FrameworkPagingDtoInterface {

    /** APIユーザ検索フラグ */
    isPartnerApiSearch: boolean;

    /** 運営者検索フラグ */
    isManagerSearch: boolean;

    /** 管理者検索フラグ */
    isAdminSearch: boolean;

    /** 検索語 */
    searchNaturalWords: string;
}


class SearchRiyoushaAllCapsuleDto extends FrameworkPagingDto implements SearchRiyoushaAllCapsuleDtoInterface {

    /** APIユーザ検索フラグ */
    isPartnerApiSearch: boolean;

    /** 運営者検索フラグ */
    isManagerSearch: boolean;

    /** 管理者検索フラグ */
    isAdminSearch: boolean;

    /** 検索語 */
    searchNaturalWords: string;

    constructor() {
        super();

        // 初期データ
        const INIT_STRING: string = "";
        const INIT_BOOLEAN: boolean = true;

        this.isPartnerApiSearch = INIT_BOOLEAN;
        this.isManagerSearch = INIT_BOOLEAN;
        this.isAdminSearch = INIT_BOOLEAN;
        this.searchNaturalWords = INIT_STRING;


    }

}

export { type SearchRiyoushaAllCapsuleDtoInterface, SearchRiyoushaAllCapsuleDto }